src/Jobs/AppendViewToSheet.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Maatwebsite\Excel\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Maatwebsite\Excel\Writer;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use Maatwebsite\Excel\Concerns\FromView;
+use PhpOffice\PhpSpreadsheet\Reader\Html;
+use Maatwebsite\Excel\Files\TemporaryFile;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Maatwebsite\Excel\Files\TemporaryFileFactory;
+
+class AppendViewToSheet implements ShouldQueue
+{
+    use Queueable, Dispatchable;
+
+    /**
+     * @var TemporaryFile
+     */
+    public $temporaryFile;
+
+    /**
+     * @var string
+     */
+    public $writerType;
+
+    /**
+     * @var int
+     */
+    public $sheetIndex;
+
+    /**
+     * @var FromView
+     */
+    public $sheetExport;
+
+    /**
+     * @var TemporaryFileFactory
+     */
+    protected $temporaryFileFactory;
+
+    /**
+     * @param FromView        $sheetExport
+     * @param TemporaryFile $temporaryFile
+     * @param string        $writerType
+     * @param int           $sheetIndex
+     * @param array         $data
+     */
+    public function __construct(FromView $sheetExport, TemporaryFile $temporaryFile, string $writerType, int $sheetIndex)
+    {
+        $this->sheetExport          = $sheetExport;
+        $this->temporaryFile        = $temporaryFile;
+        $this->writerType           = $writerType;
+        $this->sheetIndex           = $sheetIndex;
+        $this->temporaryFileFactory = app(TemporaryFileFactory::class);
+    }
+
+    /**
+     * @param Writer $writer
+     *
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     * @throws \PhpOffice\PhpSpreadsheet\Reader\Exception
+     */
+    public function handle(Writer $writer)
+    {
+        $writer = $writer->reopen($this->temporaryFile, $this->writerType);
+
+        $temporaryFile = $this->temporaryFileFactory->makeLocal();
+        $temporaryFile->put($this->sheetExport->view()->render());
```

Comment:
Is it really necessary to duplicate all this logic into here? Why can't we re-use the already existing logic?

Reply:
It seems you only have to pass the correct sheet index to `fromView()` method as new parameter. If you sheetIndex given, use the last sheet (like currently done with the sheet count)

Reply:
Thanks for your guide. I will fix it.

Reply:
Thanks :)